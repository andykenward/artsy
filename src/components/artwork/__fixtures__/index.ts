import { ArtworkQuery } from '../../../generated/graphql';

export const ARTWORK_RESULT_DATA: ArtworkQuery = {
  artwork: {
    id: 'QXJ0d29yazpob3NzYW0tZGlyYXItbmVmZXJ0aXRpLTk=',
    slug: 'hossam-dirar-nefertiti-9',
    artistNames: 'Hossam Dirar',
    imageTitle: 'Hossam Dirar, ‘Nefertiti’, 2018',
    href: 'https://artsy.net',
    image: {
      placeholder: '100.40816326530613%',
      orientation: 'landscape',
      resized: {
        url:
          'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=799&height=803&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FCbM3QgRJTqffp_SwZV5FxQ%2Flarge.jpg',
        width: 400,
        height: 803,
        __typename: 'ResizedImageUrl',
      },
      resizedRetina: {
        url:
          'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=799&height=803&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FCbM3QgRJTqffp_sdsdsSwZV5FxQ%2Flarge.jpg',
        width: 799,
        height: 803,
        __typename: 'ResizedImageUrl',
      },
      __typename: 'Image',
    },
    __typename: 'Artwork',
  },
};

export const ARTWORK_RESULT_ERROR: { errors: any; data: ArtworkQuery } = {
  errors: [
    {
      message: 'error',
      locations: [{ line: 3, column: 5 }],
      path: ['viewer', 'artwork'],
      nodes: undefined,
      source: undefined,
      positions: undefined,
      originalError: undefined,
      extensions: undefined,
      name: 'name',
    },
  ],
  data: { artwork: null },
};
