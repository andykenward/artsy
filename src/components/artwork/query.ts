import { DocumentNode } from 'graphql';
import gql from 'graphql-tag';

export const QUERY_ARTWORK: DocumentNode = gql`
  query Artwork($id: String!) {
    artwork(id: $id) {
      id
      slug
      artistNames
      imageTitle
      href
      image {
        placeholder
        orientation
        resized(width: 600) {
          url
          width
          height
        }
        resizedRetina: resized(width: 1200) {
          url
          width
          height
        }
      }
    }
  }
`;
