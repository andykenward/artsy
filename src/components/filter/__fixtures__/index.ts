import { FilterArtworksQuery } from '../../../generated/graphql';

export const FILTER_RESULT_DATA_EMPTY: FilterArtworksQuery = {
  viewer: {
    filterArtworks: {
      artworks: {
        pageInfo: {
          hasNextPage: true,
          endCursor: 'YXJyYXljb25uZWN0aW9uOjI5',
          __typename: 'PageInfo',
        },
        edges: [],
        __typename: 'ArtworkConnection',
      },
      __typename: 'FilterArtworks',
    },
    __typename: 'Viewer',
  },
};

export const FILTER_RESULT_DATA: FilterArtworksQuery = {
  viewer: {
    filterArtworks: {
      artworks: {
        pageInfo: {
          hasNextPage: true,
          endCursor: 'YXJyYXljb25uZWN0aW9uOjI5',
          __typename: 'PageInfo',
        },
        edges: [
          {
            node: {
              id: 'QXJ0d29yazpvbGdhLWtpc3NlbGV2YS1jb250cmUtdGVtcHMtMg==',
              href: '/artwork/olga-kisseleva-contre-temps-2',
              title: 'Contre temps',
              imageTitle: 'Olga Kisseleva, ‘Contre temps’, 2017',
              artistNames: 'bob',
              image: {
                placeholder: '44.83241854490249%',
                resized: {
                  url:
                    'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=179&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F_OuIfBeAhYWEZ6qGRg2EZg%2Flarge.jpg',
                  __typename: 'ResizedImageUrl',
                },
                __typename: 'Image',
              },
              __typename: 'Artwork',
            },
            __typename: 'ArtworkEdge',
          },
          {
            node: {
              id:
                'QXJ0d29yazptYXJ5LXJlaWQta2VsbGV5LXdpdGgtcGF0cmljay1rZWxsZXktcGludXBzLWluLXRoZS1idW5rcy1vZi10aGUtdXNzLXN0dXJnZW9u',
              href:
                '/artwork/mary-reid-kelley-with-patrick-kelley-pinups-in-the-bunks-of-the-uss-sturgeon',
              title: 'Pinups in the Bunks of the USS Sturgeon',
              imageTitle:
                'Mary Reid Kelley with Patrick Kelley, ‘Pinups in the Bunks of the USS Sturgeon’, 2017',
              artistNames: 'bob',
              image: {
                placeholder: '80%',
                resized: {
                  url:
                    'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=320&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FxTIMkrTaxJtqfr1T7W-bJg%2Flarge.jpg',
                  __typename: 'ResizedImageUrl',
                },
                __typename: 'Image',
              },
              __typename: 'Artwork',
            },
            __typename: 'ArtworkEdge',
          },
          {
            node: {
              id: 'QXJ0d29yazpsdWlzLW1lZGluYS1yZWQtZmllbGQtaWlp',
              href: '/artwork/luis-medina-red-field-iii',
              title: 'Red Field III',
              imageTitle: 'Luis Medina, ‘Red Field III’, 2015',
              artistNames: 'bob',
              image: {
                placeholder: '122.72727272727273%',
                resized: {
                  url:
                    'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=490&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FYAITzBZh36ozLDeq4ztmqA%2Flarge.jpg',
                  __typename: 'ResizedImageUrl',
                },
                __typename: 'Image',
              },
              __typename: 'Artwork',
            },
            __typename: 'ArtworkEdge',
          },
          {
            node: {
              id: 'QXJ0d29yazpvc28tbGMtc2luLXRpdHVsby0xOA==',
              href: '/artwork/oso-lc-sin-titulo-18',
              title: 'SIN TITULO (18)',
              imageTitle: 'OSO LC, ‘SIN TITULO (18)’, 2019',
              artistNames: 'bob',
              image: {
                placeholder: '131.50984682713346%',
                resized: {
                  url:
                    'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=526&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F8ts6ggLPfDGUEs7k1Hljbw%2Flarge.jpg',
                  __typename: 'ResizedImageUrl',
                },
                __typename: 'Image',
              },
              __typename: 'Artwork',
            },
            __typename: 'ArtworkEdge',
          },
          {
            node: {
              id:
                'QXJ0d29yazptYXJ5LXJlaWQta2VsbGV5LXdpdGgtcGF0cmljay1rZWxsZXktZ2F1ZHktbmlnaHQ=',
              href: '/artwork/mary-reid-kelley-with-patrick-kelley-gaudy-night',
              title: 'Gaudy Night',
              imageTitle:
                'Mary Reid Kelley with Patrick Kelley, ‘Gaudy Night’, 2017',
              artistNames: 'bob',
              image: {
                placeholder: '66.73913043478261%',
                resized: {
                  url:
                    'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=266&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FEeBUZRYDj56abPARNdwzyA%2Flarge.jpg',
                  __typename: 'ResizedImageUrl',
                },
                __typename: 'Image',
              },
              __typename: 'Artwork',
            },
            __typename: 'ArtworkEdge',
          },
          {
            node: {
              id: 'QXJ0d29yazptd2FtYmEtbXVsYW5nYWxhLXNlbnN1YWwtc25hcmU=',
              href: '/artwork/mwamba-mulangala-sensual-snare',
              title: 'Sensual Snare',
              imageTitle: 'Mwamba Mulangala, ‘Sensual Snare’, 2016',
              artistNames: 'bob',
              image: {
                placeholder: '218.57923497267763%',
                resized: {
                  url:
                    'https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=400&height=874&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FyML5R5JvfJXc-uGD2DLvLQ%2Flarge.jpg',
                  __typename: 'ResizedImageUrl',
                },
                __typename: 'Image',
              },
              __typename: 'Artwork',
            },
            __typename: 'ArtworkEdge',
          },
        ],
        __typename: 'ArtworkConnection',
      },
      __typename: 'FilterArtworks',
    },
    __typename: 'Viewer',
  },
};
