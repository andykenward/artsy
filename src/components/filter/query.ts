import { DocumentNode } from 'graphql';
import gql from 'graphql-tag';

export const QUERY_FILTER: DocumentNode = gql`
  query FilterArtworks($color: String, $page: Int) {
    artworksConnection(color: $color, page: $page, first: 30, after: "") {
      pageInfo {
        hasNextPage
        endCursor
      }
      id
      edges {
        cursor
        node {
          id
          href
          title
          artistNames
          imageTitle
          image {
            placeholder
            resized(width: 400) {
              url
            }
          }
        }
      }
    }
  }
`;
