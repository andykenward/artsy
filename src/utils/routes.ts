import queryString from 'query-string';
import { isNil, pick, pickBy } from 'ramda';

const allowedFilters = ['color'];

interface Filter {
  color?: string;
}

export const validateFilterObject = (
  searchQuery: string | undefined
): Filter | null => {
  if (!searchQuery || searchQuery === '') return null;
  const filter = pick(allowedFilters, queryString.parse(searchQuery));
  return pickBy((v) => !isNil(v), filter);
};

export const getFilter = (
  searchQuery: string | undefined
): { color?: string } | null => {
  return validateFilterObject(searchQuery);
};
