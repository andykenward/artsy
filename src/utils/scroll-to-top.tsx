import { RouteComponentProps } from '@reach/router';
import React, { useEffect } from 'react';

export const ScrollToTop: React.FC<RouteComponentProps> = ({
  children,
  location,
}) => {
  const { pathname } = location!;
  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname]);

  return <>{children}</>;
};
