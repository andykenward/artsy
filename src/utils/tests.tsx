import { injectGlobalStyles, Theme } from '@artsy/palette';
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from '@reach/router';
import { render } from '@testing-library/react';
import React from 'react';

export const { GlobalStyles } = injectGlobalStyles(`
  @import url("https://webfonts.artsy.net/all-webfonts.css")
`);

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(
    <Theme>
      <>
        <GlobalStyles />
        {element}
      </>
    </Theme>
  );

export const renderWithRouter = (
  element: React.ReactElement<any>,
  {
    route = '/',
    history = createHistory(createMemorySource((route = '/'))),
  } = {}
) => ({
  ...render(<LocationProvider history={history}>{element}</LocationProvider>),
  // adding `history` to the returned utilities to allow us
  // to reference it in our tests (just try to avoid using
  // this to test implementation details).
  history,
});
