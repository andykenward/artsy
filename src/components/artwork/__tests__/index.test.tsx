import { MockedResponse, wait } from '@apollo/react-testing';
import { act } from '@testing-library/react';
import React from 'react';
import { Artwork } from '..';
import { useArtworkQuery } from '../../../generated/graphql';
import { renderWithApp } from '../../../utils/tests';
import { QUERY_ARTWORK } from '../query';
import { ARTWORK_RESULT_DATA, ARTWORK_RESULT_ERROR } from '../__fixtures__';

/**
 * FYI GraphQL Fragments do not currently work in Queries for testing
 */

describe('<Artwork />', () => {
  const SLUG = 'hossam-dirar-nefertiti-9';
  const SLUG_FAIL = 'fail';
  const MOCKS: MockedResponse[] = [
    {
      request: {
        query: QUERY_ARTWORK,
        variables: { id: SLUG },
      },
      result: {
        data: ARTWORK_RESULT_DATA,
      },
    },
  ];

  it('should handle Artwork query', async () => {
    const Component = () => {
      const { data, loading } = useArtworkQuery({ variables: { id: SLUG } });
      if (!loading) {
        expect(data).toEqual(ARTWORK_RESULT_DATA);
      }
      return null;
    };
    act(() => {
      renderWithApp(<Component />, MOCKS);
    });
  });

  it('should render loading state initially', async () => {
    await act(async () => {
      const root = renderWithApp(<Artwork slug={SLUG} />, MOCKS);
      expect(root.container.firstChild).toMatchSnapshot();
    });
  });

  it('should render success response', async () => {
    await act(async () => {
      const root = renderWithApp(<Artwork slug={SLUG} />, MOCKS);
      await wait(0); // wait for response
      expect(root.container.firstChild).toMatchSnapshot();
    });
  });

  it('should render empty success response', async () => {
    const MOCKS: MockedResponse[] = [
      {
        request: {
          query: QUERY_ARTWORK,
          variables: { id: SLUG_FAIL },
        },
        result: {
          data: ARTWORK_RESULT_ERROR.data,
        },
      },
    ];
    await act(async () => {
      const root = renderWithApp(<Artwork slug={SLUG_FAIL} />, MOCKS);
      await wait(0); // wait for response
      expect(root.container.firstChild).toMatchSnapshot();
      // TODO check dom for text "no matching artwork"
    });
  });

  it.todo('error state');
});
