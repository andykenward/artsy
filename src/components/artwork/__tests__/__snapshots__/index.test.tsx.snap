// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Artwork /> should render empty success response 1`] = `
.c3 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 16px;
  line-height: 26px;
}

.c4 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
}

.c0 {
  min-height: 100vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1rem;
}

.c2 {
  display: inline-block;
  margin-bottom: 1rem;
}

@media screen and (min-width:768px) {

}

<main
  class="c0"
>
  <section
    class="c1"
  >
    <a
      class="c2"
      href="/collect"
    >
      <div
        class="c3 Typography-sc-1pie2ao-1"
        font-family="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
        font-size="16px"
      >
        &lt; Collect
      </div>
    </a>
    <div
      class="c4 Typography-sc-1pie2ao-1"
      font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
      font-size="18px"
    >
      No matching Artwork
    </div>
  </section>
</main>
`;

exports[`<Artwork /> should render loading state initially 1`] = `
.c3 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 16px;
  line-height: 26px;
}

.c4 {
  -webkit-animation: dDtqwY 1s infinite linear;
  animation: dDtqwY 1s infinite linear;
  position: absolute;
  background: #000;
  width: 25px;
  height: 6px;
  top: calc(50% - 6px / 2);
  left: calc(50% - 25px / 2);
}

.c0 {
  min-height: 100vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1rem;
}

.c2 {
  display: inline-block;
  margin-bottom: 1rem;
}

@media screen and (min-width:768px) {

}

<main
  class="c0"
>
  <section
    class="c1"
  >
    <a
      class="c2"
      href="/collect"
    >
      <div
        class="c3 Typography-sc-1pie2ao-1"
        font-family="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
        font-size="16px"
      >
        &lt; Collect
      </div>
    </a>
    <div
      class="c4"
      color="black100"
      height="6"
      width="25"
    />
  </section>
</main>
`;

exports[`<Artwork /> should render success response 1`] = `
.c3 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 16px;
  line-height: 26px;
}

.c6 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
}

.c7 {
  color: #000;
  -webkit-transition: color 0.25s;
  transition: color 0.25s;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #000;
}

.c0 {
  min-height: 100vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1rem;
}

.c2 {
  display: inline-block;
  margin-bottom: 1rem;
}

.c5 {
  max-width: 100%;
  height: auto;
  max-height: 58vh;
  object-fit: contain;
}

.c4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0;
  padding: 0;
}

@media screen and (min-width:768px) {

}

<main
  class="c0"
>
  <section
    class="c1"
  >
    <a
      class="c2"
      href="/collect"
    >
      <div
        class="c3 Typography-sc-1pie2ao-1"
        font-family="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
        font-size="16px"
      >
        &lt; Collect
      </div>
    </a>
    <figure
      class="c4"
    >
      <img
        alt=""
        class="c5"
        height="602"
        src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=600&height=602&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FCbM3QgRJTqffp_SwZV5FxQ%2Flarge.jpg"
        srcset="https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=600&height=602&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FCbM3QgRJTqffp_SwZV5FxQ%2Flarge.jpg, https://d7hftxdivxxvm.cloudfront.net?resize_to=fit&width=1200&height=1204&quality=80&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FCbM3QgRJTqffp_SwZV5FxQ%2Flarge.jpg 2x"
        width="600"
      />
      <figcaption
        style="max-width: 600px;"
      >
        <div
          class="c6 Typography-sc-1pie2ao-1"
          font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          font-size="18px"
        >
          <strong>
            Hossam Dirar
          </strong>
        </div>
        <div
          class="c6 Typography-sc-1pie2ao-1"
          font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          font-size="18px"
        >
          Hossam Dirar, ‘Nefertiti’, 2018
        </div>
        <a
          class="c7"
          href="https://www.artsy.net/artwork/hossam-dirar-nefertiti-9"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="c6 Typography-sc-1pie2ao-1"
            font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
            font-size="18px"
          >
            Source
          </div>
        </a>
      </figcaption>
    </figure>
  </section>
</main>
`;
